src/blockchain/rpc.rs

```diff
@@ -294,6 +295,38 @@ impl WalletSync for RpcBlockchain {
         }) {
             let txid = tx_result.info.txid;
             list_txs_ids.insert(txid);
+            if tx_result.detail.category == GetTransactionResultDetailCategory::Immature {
+                // Immature UTXOs don't appear in rpc's listunspent, so we
+                // have to manually add them to the current_utxos list
+                // FIXME: This will produce a wrong result if the immature utxo is spent,
+                // which *should* be impossible.
+                let utxo = ListUnspentResultEntry {
+                    txid: tx_result.info.txid,
+                    vout: tx_result.detail.vout,
+                    amount: tx_result
+                        .detail
+                        .amount
+                        .to_unsigned()
+                        .expect("immature txs amount can't be negative"),
+                    script_pub_key: tx_result
+                        .detail
+                        .address
+                        .as_ref()
+                        .expect("Should always be here")
```

Comment:
Some descriptors are encoded as scripts that don't have an address form, so I don't think it's safe to expect here